config/always-compiler-options.m4

```diff
@@ -181,6 +181,32 @@ AC_DEFUN([ZFS_AC_CONFIG_ALWAYS_CC_NO_CLOBBERED], [
 	AC_SUBST([NO_CLOBBERED])
 ])
 
+dnl #
+dnl # Check if gcc supports -Wno-cast-function-type option.
```

Comment:
You probably want to s/gcc/$CC/ in much of the comment since the whole reason for the issue is that many versions of clang barf on it.

Reply:
but, oddly, not the one I'm building this with on my dev system. updated